tools/format/format.html

```diff
@@ -0,0 +1,47 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>Cataclysm web linting tool</title>
+    <style>
+      * { font-family: 'monospace'; }
+      input { display: block; }
+    </style>
+    <script>
+      function lint(elem) {
+          if (elem.value.length == 0) {
+              return false;
+          }
+          try {
+              JSON.parse(elem.value);
+          } catch (e) {
+              alert('Syntax error');
+              return false;
+          }
+
+          var xhr = new XMLHttpRequest();
+          xhr.open('POST', '/tools/format/format.pl');
+          xhr.setRequestHeader('Content-Type', 'application/x-www-form-urlencoded');
+          xhr.onload = function() {
+              if (xhr.status == 200) {
+                  elem.value = xhr.responseText;
+              } else if (xhr.status == 304) {
+                  // no-op
+              } else if (xhr.status == 400) {
+                  alert(xhr.responseText);
+              } else {
+                  alert('Linter currently unavailable');
+              }
+          };
+          xhr.send(encodeURI('data=' + elem.value));
```

Comment:
This needs to be more like `xhr.send('data=' + encodeURIComponent(elem.value));` -- any of `, / ? : @ & = + $ #` should be URL-encoded, which is what `encodeURIComponent` does (`encodeURI` leaves them alone).